Ports/acpica-tools/patches/0004-Remove-unsupported-warning-flags-for-Clang-support.patch

```diff
@@ -0,0 +1,38 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Gurkirat Singh <tbhaxor@gmail.com>
+Date: Fri, 18 Aug 2023 08:17:35 +0530
+Subject: [PATCH] Remove unsupported warning flags for Clang support
```

Comment:
We should still add these flags for GCC. Clang warns by default about two of those (at least if I remember correctly), but I don't think GCC does.

Reply:
Yes, GCC doesn't warn. I do not know how to exclude this patch if SERENITY_TOOLCHAIN is GNU, please guide me.

Reply:
```
if [ $SERENITY_TOOLCHAIN = 'Clang' ]; then
    export CFLAGS="-Wno-logical-op -Wno-missing-parameter-type -Wno-old-style-declarations "
fi
```

.. assuming the makefile puts CFLAGS after CWARNINGFLAGS anyway. If it doesn't then some way to add those flags to the end of the final CFLAGS used to compile would be correct.

Reply:
given this https://github.com/acpica/acpica/blob/master/generate/unix/Makefile.config#L147

I would expect adding `OPT_CFLAGS=""` to makeopts to completely disable all the warnings.

Reply:
> I would expect adding OPT_CFLAGS="" to makeopts to completely disable all the warnings.

If will then overwrite flags coming from `CWARNINGFLAGS`.

Reply:
```make
# OPT_CFLAGS can be overridden on the make command line by
#   adding OPT_CFLAGS="..." to the invocation.
```

Reply:
I implemented a feature to get patch files based on conditions. Please check.

Reply:
Once you approve this change, I will also add the documentation and squash commits.

Reply:
Let's not start with making selected patches depend on the toolchain. This potentially also has effects on generating the README as well as our `dev` mode.

Either change the patch itself to let the Makefile detect the toolchain, move the affected flags into a separately enabled/disabled option, or just override the selection of warning flags wholesale by exporting `OPT_CFLAGS` (with only the flags that we _do_ want) in `package.sh`.

Reply:
I will work on it. Now I am thinking, your words make sense.

Reply:
I am using different approach, it will still apply the patch now using [`override`](https://www.gnu.org/software/make/manual/make.html#Override-Directive) directive in the patched Makefiles so that we can pass stripped warning flags in GNU toolchain.

Reply:
Ping @timschumi

Reply:
Upon checking, it seems like the addition of `override` discards all subsequently added flags, which is most likely not what we want.

I uploaded a different, leaner fix as #20894, which seems to sufficiently solve the issue as well. Feel free to test and let me know whether it works for you as well.

Reply:
This is what override does

![image](https://github.com/SerenityOS/serenity/assets/28386721/00b118f4-3085-4f17-bfa4-5db5e38ff509)

You can try this code on your system

```make
VAR1 += hello\
        world
override VAR2 += hello\
                world

print:
        @echo $(VAR1)
        @echo $(VAR2)
```

Reply:
Since `CWARNINGFLAGS` is the variable that holds all these flags, so I removed them in general and for GNU toolchain, I added these flags.

Only difference what it will make is that the value passed via make cli invocation in overriden variables will be **prepended**. I don't think so it will make a difference to gcc, because `gcc -Werror -Wall` is same as `gcc -Wall -Werror`.

Reply:
I know what `override` does and what your intention behind adding it is. That doesn't change the fact that adding `override` to the first assignment renders the following assignments ineffective.

Please test the PR that I linked. If that PR works for you we'll go ahead with that, because a 60+ line patch file isn't really proportional to the problem that we are trying to solve (and neither is adding `override` to all the other assignments).